{% extends "layout.html" %}
{% block body %}
<div class="container">
    <div id="about" class="row card">
        <div class="col-md-8 col-md-offset-2">
            <h1>About FLOSsies’ inspiration, participants, and sequencing.</h1>
        </div>
        {% include 'logos.html' %}
    </div>

        <div class="col-md-12">
            <p style="text-align: left">
The FLOSSIES sequencing project was inspired by the need for more information about the frequency of genomic variants from appropriate controls for breast cancer genetics studies. Existing public databases are large, but not stratified by sex or age. The FLOSSIES project includes allele frequencies of all classes of variants at known and candidate genes among women who are cancer free and older than 70. The participants are Fabulous Ladies Over Seventy (FLOSsies).</p>

<h3>Participants</h3>

<p>FLOSsie participants are women who joined the <a href="https://www.whi.org/SitePages/WHI%20Home.aspx">Women’s Health Initiative</a> (WHI) between 1993 and 2005. All participants were 50-79 years old and had no history of cancer at enrollment. More than 160,000 women enrolled in WHI, contributed DNA at enrollment, and have been followed ever since. From the WHI participants who are now older than 70 and have remained cancer-free, 10,000 women were selected at random for this sequencing project: 7,000 who self-identified as European American and 3,000 who self-identified as African American. Ancestries were confirmed with 500 ancestry-informative markers.</p>

<h3>Genes</h3>

<p>Genes analyzed include established breast cancer genes, both high and moderate penetrance, and genes that have been suggested to predispose to breast cancer when mutant. Strength of evidence varies for these candidate genes. The genes are:</p>

<table>
    <tr>
        <td>BRCA1</td>
        <td>BRCA2</td>
        <td>ATM</td>
        <td>ATR</td>
    </tr><tr>
        <td>BAP1</td>
        <td>BARD1</td>
        <td>BRIP1</td>
        <td>CDH1</td>
    </tr><tr>
        <td>CHEK1</td>
        <td>CHEK2</td>
        <td>CTNNA1</td>
        <td>FAM175A</td>
    </tr><tr>
        <td>FANCM</td>
        <td>GEN1</td>
        <td>MRE11A</td>
        <td>NBN</td>
    </tr><tr>
        <td>PALB2</td>
        <td>PTEN</td>
        <td>RAD51B</td>
        <td>RAD51C</td>
    </tr><tr>
        <td>RAD51D</td>
        <td>RECQL</td>
        <td>RINT 1</td>
        <td>SLX4</td>
    </tr><tr>
        <td>STK11</td>
        <td>TP53</td>
        <td>XRCC2
    </tr>
</table>

<p>For each gene, rare and common variants in coding regions, 5’UTR, 3’UTR, donor splice sites with 6 flanking bp, and acceptor splice sites with 20 flanking bp are included. Point mutations, small indels, and CNVs are included (although CNV calls are still incomplete in v1).</p>

<h3>Sequencing</h3>

<p>Sequencing was carried out in the <a href="http://www.gs.washington.edu/faculty/king.htm">King Lab</a> at the <a href="http://www.gs.washington.edu/">University of Washington</a>, Seattle, and by <a href="https://www.color.com/">Color Genomics</a> on an Illumina HiSeq with 2x100 bp paired-end reads (King lab) or NextSeq with 2x150 bp paired-end reads (Color) using modified versions of the BROCA panel (Walsh et al. PNAS 107:12629-33, 2010). Median coverage per sample was >250x.</p>

<p>Paired-end sequence reads were aligned to the human reference genome (hg19) using Burrows-Wheeler Aligner 0.7.9a. Removal of PCR duplicates, sorting, and indexing were carried out with SAMtools v0.1.19. Data was excluded for 3 samples with low coverage and 2 samples with low quality. For quality control, WHI included 105 duplicates and 20 triplicates, blinded to the sequencing teams. These duplicates and triplicates were all identified and removed.</p>

<p>Indel realignments and base quality score recalibration were based on with Genome Analysis Tool Kit (GATK; v3.0-0) using recommended parameters. Variants were detected with GATK Unified Genotyper. Variants were included if variant fraction was at least 0.25. Copy number variants were identified using our read-depth-based in-house pipeline.</p>

<h3>Ancestry confirmation</h3>

<p>Ancestry of each sample was confirmed using 500 ancestry-informative markers and principal components analysis (Wang C et al. Am J Hum Genet 96:926-37, 2015), calibrating to genotypes from Human Genome Diversity Panel (Li JZ et al. Science 319:1100-04, 2008).</p>

<p>Each sample was assigned to the population with highest estimated likelihood of ancestry.</p>

<p>Ancestry analysis yielded estimates of 7325 women of European American ancestry and 2559 women of African American ancestry. 14 samples from women of South Asian or East Asian ancestry were excluded.
            </p>
        </div>
    </div>

    <div id="exac" class="row card">
        <div class="col-md-8 col-md-offset-2">
            <br />
            <p>This site uses <a href="https://github.com/ColorGenomics/exac_browser">a modified version</a> of the open source <a href="https://github.com/konradjk/exac_browser">ExAC Browser</a>, developed by <a href="http://broadinstitute.org/">The Broad Institute</a> and distributed <a href="https://github.com/konradjk/exac_browser/blob/master/LICENSE">under the MIT License</a>.
            </p>
        </div>
    </div>
</div>
{% endblock %}
