{% extends "layout.html" %}
{% block body %}
    <div class="container" style="font-size: 16px;">
        <h1>Frequently Asked Questions</h1>
        <h2>General</h2>
        <ul class="media-list">
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">How should I cite discoveries made using ExAC data?</h4>
                    Please cite the ExAC flagship paper available <a href="http://dx.doi.org/10.1038/nature19057">here</a>.
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">I have identified a rare variant in ExAC that I believe is associated with a specific clinical phenotype. What phenotype data are available for these individuals?</h4>
                    Many of the individuals who have contributed data to ExAC were not fully consented for phenotype data sharing, and unfortunately at this time we are typically unable to provide any information about the clinical status of variant carriers. We have made every effort to exclude individuals with severe pediatric diseases from the ExAC data set, and certainly do not expect our data set to be <i>enriched</i> for such individuals, but we typically cannot rule out the possibility that some of our participants do actually suffer from your disease of interest.
                    <br><br>
                    In the future it may be possible to obtain phenotype data for a subset of ExAC samples. If you are interested in using such data in your studies, please <a href="mailto:exomeconsortium@gmail.com">email us</a>.
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">Can I get access to individual-level genotype data from ExAC?</h4>
                    Many of the samples in ExAC have individual-level sequencing data deposited in <a href="http://www.ncbi.nlm.nih.gov/gap">dbGaP</a>, and these can be accessed by <a href="https://dbgap.ncbi.nlm.nih.gov/aa/wga.cgi?page=login">applying through that repository</a>. However, many other samples come from unpublished cohorts or from samples not in dbGaP. There is not currently any mechanism to systematically obtain individual-level genotype data from the database as a whole.
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">What are the restrictions on data usage?</h4>
                    All data here are released under a <a href="http://www.wellcome.ac.uk/stellent/groups/corporatesite/@policy_communications/documents/web_document/wtd003207.pdf">Fort Lauderdale Agreement</a> for the benefit of the wider biomedical community. You can freely download and search the data, and use it for publications focused on specific sets of variants (for instance, assessing the frequency of a set of candidate causal variants observed in a collection of rare disease patients). However, we ask that you not publish global (genome-wide) analyses of these data until after the ExAC flagship paper has been published, estimated to be in early 2015. If youâ€™re uncertain which category your analyses fall into, please <a href="mailto:exomeconsortium@gmail.com">email us</a>. The data are available under the <a href="http://opendatacommons.org/licenses/odbl/1.0/">ODC Open Database License (ODbL)</a> (summary available <a href="http://www.opendatacommons.org/licenses/odbl/1-0/summary/">here</a>): you are free to share and modify the ExAC data so long as you attribute any public use of the database, or works produced from the database; keep the resulting data-sets open; and offer your shared or adapted version of the dataset under the same ODbL license.
                </div>
            </li>
        </ul>
        <h2>Constraint</h2>
        <ul class="media-list">
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">How was the expected number of variants determined?</h4>
                    We used a depth corrected probability of mutation for each gene to predict the expected variant counts. More details can be found in section 4.1 of the supplement in <a href="http://www.nature.com/nature/journal/v536/n7616/full/nature19057.html">Lek et al</a>. Note that the expected variant counts for exons with a median depth < 1 were removed from the totals.
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">Which variants are included in the observed counts?</h4>
                    We included single nucleotide changes that occurred in the canonical transcript and had 123 or fewer alternative alleles and a VQSLOD >= -2.632. The counts represent the number of unique variants and not the allele count of said variants. Observed variant counts in exons with a median depth < 1 were removed from the totals.
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">Why are there fewer variants in the constraint table than depicted on the gene page?</h4>
                    We only included variants that were found in the canonical transcript of the gene. On the gene page, variants found in all transcripts are depicted. Additionally, both observed and expected variant counts were removed for exons with a median depth < 1.
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">What is included in LoF?</h4>
                    Nonsense, splice acceptor, and splice donor variants caused by single nucleotide changes.
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">What are the constraint metrics?</h4>
                    <p>For synonymous and missense, we created a signed Z score for the deviation of observed counts from the expected number. Positive Z scores indicate increased constraint (intolerance to variation) and therefore that the gene had fewer variants than expected. Negative Z scores are given to genes that had a more variants than expected.</p>
                    <p>For LoF, we assume that there are three classes of genes with respect to tolerance to LoF variation: null (where LoF variation is completely tolerated), recessive (where heterozygous LoFs are tolerated), and haploinsufficient (where heterozygous LoFs are not tolerated). We used the observed and expected variants counts to determine the probability that a given gene is extremely intolerant of loss-of-function variation (falls into the third category). The closer pLI is to one, the more LoF intolerant the gene appears to be. We consider pLI >= 0.9 as an extremely LoF intolerant set of genes.</p>
                    <p>More details can be found in the supplement of <a href="http://biorxiv.org/content/early/2015/10/30/030338">Lek et al</a>.</p>
                </div>
            </li>
        </ul>
        <h2>Copy Number Variation (CNV)</h2>
        <ul class="media-list">
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">How were CNVs discovered?</h4>
                    Deletions and duplications were called based on read depth using XHMM (<a href="http://www.sciencedirect.com/science/article/pii/S000292971200417X">Fromer et al.</a>). Explicit methodological details are provided in <a href="http://www.nature.com/ng/journal/vaop/ncurrent/full/ng.3638.html">Ruderfer et al</a>.
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">How were CNV intolerance scores calculated?</h4>
                    It is not straightforward to model genic CNV rates expected under neutrality in a direct manner, as can be done for SNVs using trinucleotide mutation rates and the s known sequence. To derive expected values, we therefore fit a linear regression model for the observed CNV rate per gene based on gene length, coding sequence length, number of targets, GC content, sequence complexity, genomic localization within pairs of segmental duplications, and sequencing read depth. Intolerances scores were calculated as the normalized and winsorized model residuals, negated such that higher positive values indicate greater intolerance (a lower than expected rate of CNVs for that gene).
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">What are the limitations of these data?</h4>
                    We note several limitations of these data: 1) Making accurate CNV calls from read depth data, in particular from targeted short read sequencing, is difficult and combined with the rarity of these events will add noise to estimates of frequencies and intolerance scores. 2) Complex CNVs such as multi-allelic events or common CNVs have been removed due to methodological limitations. 3) Mutation rates for CNVs are not well quantified making particular regions of known recurrence, such as those within pairs of tandem segmental duplications, more difficult to accurately assess intolerance. 4) Since CNVs often span multiple genes intolerance scores will be correlated over genes nearby.
                </div>
            </li>
        </ul>
        <h2>Technical details</h2>
        <ul class="media-list">
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">What genome build is the ExAC data based on?</h4>
                    All data are based on <a href="ftp://ftp.broadinstitute.org/pub/seq/references/Homo_sapiens_assembly19.fasta">GRCh37/hg19</a>
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">What version of dbSNP was used to annotate variants?</h4>
                    dbSNP 135. The browser supports RSIDs up to dbSNP 141, but these are not included in the VCF.
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">What version of Gencode was used to annotate variants?</h4>
                    Version 19 (annotated with VEP version 85).
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">Why does the browser seem to disagree with the ExAC VCF at this multiallelic site?</h4>
                    Due to the limitations of the VCF format, multi-allelic variants are put together on one VCF line.
                    This inevitably adds complexity to otherwise simple variants,
                    and thus when parsing onto the browser, we apply a <a href="http://www.cureffi.org/2014/04/24/converting-genetic-variants-to-their-minimal-representation/">minimal representation</a> script.
                    For instance, a variant whose REF is GC and ALT alleles are TC,G - the first ALT allele is actually a SNP and will be represented in the browser as G->T.
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">Why does the allele frequency (AF) found in the ExAC VCF differ from the AF shown on the browser?</h4>
                    The AF shown in the VCF is AC/AN.  The browser however shows the adjusted allele frequency (AC_adj/AN_adj). The raw counts (AC and AN) refer to the total number of chromosomes with this allele, and total that were able to be called (whether reference or alternate), respectively. Thus, the allele frequency is AC/AN. However, for more stringent applications, we adjusted the allele count to only include individuals with genotype quality (GQ) >= 20 and depth (DP) >= 10, which we term AC_adj (and correspondingly, AN_adj for total number of chromosomes from individuals with those criteria regardless of genotype call).
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">Are all the individuals in the <a href="http://evs.gs.washington.edu/EVS/">Exome Variant Server</a> included?</h4>
                    No. We were not given permission from dbGaP to include individuals from several of the cohorts included in the NHLBI's Exome Sequencing Project. As a result, genuine rare variants that are present in the EVS may not be observed in ExAC.
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">Do the cancer samples in the database include tumor exomes, or is this from germline samples only?</h4>
                    All of the "cancer" samples in the current release of ExAC are blood ("germline") samples from TCGA. We excluded any sample labeled as tumor. However, note that some sample/label swaps may have occurred in TCGA; in addition, it is possible that in some patients the blood samples are contaminated by circulating tumor cells.
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">What populations are represented in the ExAC data?</h4>
                    <table border="1" cellspacing="50" cellpadding="50" width="544">
                        <tbody align="center">
                            <tr>
                                <td width="125" valign="top">
                                    <p>
                                        <strong>Population</strong>
                                    </p>
                                </td>
                                <td width="109" valign="top">
                                    <p>
                                        <strong>Male Samples</strong>
                                    </p>
                                </td>
                                <td width="113" valign="top">
                                    <p>
                                        <strong>Female Samples</strong>
                                    </p>
                                </td>
                                <td width="96" valign="top">
                                    <p>
                                        <strong>Total</strong>
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="125" valign="top">
                                    <p>
                                        African/African American (AFR)
                                    </p>
                                </td>
                                <td width="109" valign="top">
                                    <p>
                                        1,888
                                    </p>
                                </td>
                                <td width="113" valign="top">
                                    <p>
                                        3,315
                                    </p>
                                </td>
                                <td width="96" valign="top">
                                    <p>
                                        5,203
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="125" valign="top">
                                    <p>
                                        Latino (AMR)
                                    </p>
                                </td>
                                <td width="109" valign="top">
                                    <p>
                                        2,254
                                    </p>
                                </td>
                                <td width="113" valign="top">
                                    <p>
                                        3,535
                                    </p>
                                </td>
                                <td width="96" valign="top">
                                    <p>
                                        5,789
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="125" valign="top">
                                    <p>
                                        East Asian (EAS)
                                    </p>
                                </td>
                                <td width="109" valign="top">
                                    <p>
                                        2,016
                                    </p>
                                </td>
                                <td width="113" valign="top">
                                    <p>
                                        2,311
                                    </p>
                                </td>
                                <td width="96" valign="top">
                                    <p>
                                        4,327
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="125" valign="top">
                                    <p>
                                        Finnish (FIN)
                                    </p>
                                </td>
                                <td width="109" valign="top">
                                    <p>
                                        2,084
                                    </p>
                                </td>
                                <td width="113" valign="top">
                                    <p>
                                        1,223
                                    </p>
                                </td>
                                <td width="96" valign="top">
                                    <p>
                                        3,307
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="125" valign="top">
                                    <p>
                                        Non-Finnish European (NFE)
                                    </p>
                                </td>
                                <td width="109" valign="top">
                                    <p>
                                        18,740
                                    </p>
                                </td>
                                <td width="113" valign="top">
                                    <p>
                                        14,630
                                    </p>
                                </td>
                                <td width="96" valign="top">
                                    <p>
                                        33,370
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="125" valign="top">
                                    <p>
                                        South Asian (SAS)
                                    </p>
                                </td>
                                <td width="109" valign="top">
                                    <p>
                                        6,387
                                    </p>
                                </td>
                                <td width="113" valign="top">
                                    <p>
                                        1,869
                                    </p>
                                </td>
                                <td width="96" valign="top">
                                    <p>
                                        8,256
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="125" valign="top">
                                    <p>
                                        Other (OTH)
                                    </p>
                                </td>
                                <td width="109" valign="top">
                                    <p>
                                        275
                                    </p>
                                </td>
                                <td width="113" valign="top">
                                    <p>
                                        179
                                    </p>
                                </td>
                                <td width="96" valign="top">
                                    <p>
                                        454
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="125" valign="top">
                                    <p>
                                        <strong>Total</strong>
                                    </p>
                                </td>
                                <td width="109" valign="top">
                                    <p>
                                        33,644
                                    </p>
                                </td>
                                <td width="113" valign="top">
                                    <p>
                                        27,062
                                    </p>
                                </td>
                                <td width="96" valign="top">
                                    <p>
                                        <strong>60,706</strong>
                                    </p>
                                </td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">What ethnicities are represented in the "other" population?</h4>
                   Individuals were classified as "other" if they did not unambiguously cluster with the major populations (i.e. EUR, NFE, AFR, SAS, EAS and AMR) in a principal component analysis (PCA).
                </div>
            </li>
            <li class="media">
                <div class="media-body">
                    <h4 class="media-heading">What cohorts are represented in the ExAC data?</h4>
                    <table border="1" cellspacing="0" cellpadding="0">
                        <tbody align="center">
                            <tr>
                                <td width="357">
                                    <p>
                                        <strong>Consortium/Cohort</strong>
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        <strong>Samples</strong>
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        1000 Genomes
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        1,851
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        Bulgarian Trios
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        461
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        GoT2D
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        2,502
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        Inflammatory Bowel Disease
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        1,675
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        Myocardial Infarction Genetics Consortium
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        14,622
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        NHLBI-GO Exome Sequencing Project (ESP)
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        3,936
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        National Institute of Mental Health (NIMH) Controls
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        364
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        SIGMA-T2D
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        3,845
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        Sequencing in Suomi (SISu)
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        948
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        Swedish Schizophrenia &amp; Bipolar Studies
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        12,119
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        T2D-GENES
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        8,980
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        Schizophrenia Trios from Taiwan
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        1,505
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        The Cancer Genome Atlas (TCGA)
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        7,601
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        Tourette Syndrome Association International Consortium for Genomics (TSAICG)
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        297
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="357">
                                    <p>
                                        <strong>Total</strong>
                                    </p>
                                </td>
                                <td width="81">
                                    <p align="right">
                                        60,706
                                    </p>
                                </td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </li>
        </ul>
    </div>
{% endblock %}
