<script type="text/javascript">
    window.is_gene = typeof window.transcripts_in_gene != 'undefined';
</script>
<script type="text/template" id="variant-table-template">
{# WHI
{% if transcripts_in_gene is defined %}
    <div>
        <small>&dagger; denotes a consequence that is for a non-canonical transcript</small>
    </div>
{% endif %}
#}

<table id="variant_table" class="tablesorter table table-striped table-bordered table-hover">
    <caption>
        Variants
        <a id="export_to_csv" class="pull-right" href="">
            <i class="fa fa-arrow-right fa-20px"></i> Export table to CSV
        </a>
    </caption>
    <thead>
        <tr id="variant_consequence_types" class="btn-group tablesorter-ignoreRow sticky-false" data-toggle="buttons">
            <td class="btn btn-primary active consequence_display_buttons" id="consequence_button_all">
                <input type="radio">
                <nobr>All (<%= table_variants.length %>)</nobr>
            </td>
            <% _.each(consequence_counts, function(csq_count) { %>
                <td class="btn btn-primary consequence_display_buttons" id="consequence_button_<%= csq_count[0] %>">
                    <input type="radio">
                    <nobr><%= consequence_name(csq_count[0]) %> (<%= csq_count[1] %>)</nobr>
                </td>
            <% }); %>
        </tr>
        <tr>
            <th class="tooltip-table-header omit_csv" data-tooltip="chr:pos reference/alternate &#xa;(dbSNP135 rsID)">
                Variant</th>
{# WHI
            <th class='hidden-xs'>Chrom</th>
            <th class='hidden-xs'>Position</th>
#}
            <td class="hidden">RSID</td>
            <th class="hidden">Reference</th>
            <th class="hidden">Alternate</th>
            <th class="hidden tooltip-table-header" data-tooltip="HGVS annotation (protein change, where defined, or transcript change for splice variants, otherwise empty)">Consequence</th>
            <th class="hidden-xs">HGVSp</th>
            <th class="hidden-xs">HGVSc</th>
{# WHI: no filtering
            <th class="hidden-xs tooltip-table-header" data-tooltip="VQSR sensitivity filter & hard filters">Filter</th>
#}
            <th class="tooltip-table-header" data-tooltip="Variant Effect Predictor (VEP) &#xa; annotation using Gencode 81.&#xa; Worst across all transcripts of this gene.">Annotation</th>
            <th class="hidden tooltip-table-header" data-tooltip="Various flags (LoF from LOFTEE, MNP) &#xa; that may change annotation/confidence">Flags</th>
            <th class="tooltip-table-header" data-tooltip="Alternate allele count in genotypes &#xa;(genotype quality >=20 & depth >=10)">Instances</th>
            <th class="hidden tooltip-table-header" data-tooltip="Total number of called genotypes &#xa;(genotype quality >= 20 & depth >=10)">Allele Number</th>
{# WHI
            {% if chrom != 'Y' %}
                <th class="hidden-xs tooltip-table-header" data-tooltip="Number of homozygous individuals for this alternate allele">Homozygotes</th>
            {% endif %}
            {% if chrom == 'X' or chrom == 'Y' %}
                <th class="hidden-xs tooltip-table-header" data-tooltip="Number of hemizygous individuals for this alternate allele">Hemizygotes</th>
            {% endif %}
#}
            <th class="tooltip-table-header allele_frequency_column" data-tooltip="Frequency using only &#xa; high-quality genotypes &#xa; The graphic displays &#xa; allele frequency on a  &#xa; discrete scale: Singletons, &#xa; <1/10,000, <1/1000, <1%, &#xa; <5%, <50%, >50%">
                Frequency</th>
            <th class="hidden">Allele Count African</th>
            <th class="hidden">Allele Number African</th>
            <th class="hidden">Homozygote Count African</th>
            <th class="hidden">Allele Count East Asian</th>
            <th class="hidden">Allele Number East Asian</th>
            <th class="hidden">Homozygote Count East Asian</th>
            <th class="hidden">Allele Count European (Non-Finnish)</th>
            <th class="hidden">Allele Number European (Non-Finnish)</th>
            <th class="hidden">Homozygote Count European (Non-Finnish)</th>
            <th class="hidden">Allele Count Finnish</th>
            <th class="hidden">Allele Number Finnish</th>
            <th class="hidden">Homozygote Count Finnish</th>
            <th class="hidden">Allele Count Latino</th>
            <th class="hidden">Allele Number Latino</th>
            <th class="hidden">Homozygote Count Latino</th>
            <th class="hidden">Allele Count Other</th>
            <th class="hidden">Allele Number Other</th>
            <th class="hidden">Homozygote Count Other</th>
            <th class="hidden">Allele Count South Asian</th>
            <th class="hidden">Allele Number South Asian</th>
            <th class="hidden">Homozygote Count South Asian</th>
        </tr>
    </thead>
    <tbody>
    <% _.each(table_variants, function(variant) { %>
        <tr class="table_variant" id="variant_<%= variant.variant_id %>"
            category="<%= variant.category %>" consequence="<%= variant.major_consequence %>" frequency="<%= variant.allele_freq %>">
            <td class="omit_csv">
                <a href="/variant/<%= variant.chrom %>-<%= variant.pos %>-<%= variant.ref %>-<%= variant.alt %>">
                <%= variant.chrom %>:<%= variant.pos %>
                <%= variant.ref %>
                <%= variant.alt %>
                <% if (variant.rsid != '.') { %>
                    (<%= variant.rsid %>)
                <% } %>
                </a>
            </td>
{# WHI
            <td class='hidden-xs'> <%= variant.chrom %></td>
            <td class='hidden-xs'> <%= variant.pos %></td>
#}
            <td class="hidden"><%= variant.rsid %></td>
            <th class="hidden"> <%= variant.ref %></th>
            <th class="hidden"> <%= variant.alt %></th>
            <td class='hidden'>
                <% if (variant.HGVS && variant.HGVS != '') { %>
                    <% if (variant.HGVS.length > 20) { %>
                        <%= variant.HGVS.slice(0,20) %>...<% } else { %>
                        <%= variant.HGVS %><% } %>
{# WHI
                        <% if (is_gene & variant.CANONICAL != 'YES') { %><span class="tooltip-table-header" data-tooltip="Annotation is for non-canonical transcript">&dagger;</span><% } %>
#}
                <% } %>
            </td>
            <td title='<%= variant.HGVSp %>'> <%= variant.HGVSp %></td>
            <td title='<%= variant.HGVSc %>'> <%= variant.HGVSc %></td>
{# WHI: no filtering
            <td class='hidden'> <%= variant.filter %> </td>
#}
            <td class="<%= variant.category %>" data-consequence="<%= variant.major_consequence %>">
                <%= consequence_name(variant.major_consequence) %>
             </td>
            <td class="hidden">
                <% _.each(variant.flags, function(flag) { %>
                    <% if (flag == 'LoF flag') { span_tag = 'warning' } else { span_tag = 'danger' } %>
                    <a href="/variant/<%= variant.chrom %>-<%= variant.pos %>-<%= variant.ref %>-<%= variant.alt %>"><span class="label label-<%= span_tag %> tooltip-table-header" data-tooltip="Variant annotation dubious: &#xa; see variant page for details"><%= flag %></span></a>
                <% }); %></td>
            <td> <%= variant.allele_count %> </td>
            <td class='hidden'> <%= variant.allele_num %> </td>
{# WHI
            {% if chrom != 'Y' %}
                <td class="hidden-xs"> <%= variant.hom_count %> </td>
            {% endif %}
            {% if chrom == 'X' or chrom == 'Y' %}
                <td class="hidden-xs"> <%= variant.hemi_count %> </td>
            {% endif %}
#}
            <td>
                <% if (variant.allele_freq) { %>
                    <%= variant.allele_freq.toPrecision(4) %>
                <% } else { %>
                    <%= variant.allele_freq %>
                <% } %>
                <div id="variant_af_box_<%= variant.variant_id %>" style="float: right;"></div>
            </td>
            <% if (variant.pop_acs && variant.pop_ans && variant.pop_homs) { %>
                <td class="hidden"> <%= variant.pop_acs["African"] %></td>
                <td class="hidden"> <%= variant.pop_ans["African"] %></td>
                <td class="hidden"> <%= variant.pop_homs["African"] %></td>
                <td class="hidden"> <%= variant.pop_acs["East Asian"] %></td>
                <td class="hidden"> <%= variant.pop_ans["East Asian"] %></td>
                <td class="hidden"> <%= variant.pop_homs["East Asian"] %></td>
                <td class="hidden"> <%= variant.pop_acs["European (Non-Finnish)"] %></td>
                <td class="hidden"> <%= variant.pop_ans["European (Non-Finnish)"] %></td>
                <td class="hidden"> <%= variant.pop_homs["European (Non-Finnish)"] %></td>
                <td class="hidden"> <%= variant.pop_acs["European (Finnish)"] %></td>
                <td class="hidden"> <%= variant.pop_ans["European (Finnish)"] %></td>
                <td class="hidden"> <%= variant.pop_homs["European (Finnish)"] %></td>
                <td class="hidden"> <%= variant.pop_acs["Latino"] %></td>
                <td class="hidden"> <%= variant.pop_ans["Latino"] %></td>
                <td class="hidden"> <%= variant.pop_homs["Latino"] %></td>
                <td class="hidden"> <%= variant.pop_acs["Other"] %></td>
                <td class="hidden"> <%= variant.pop_ans["Other"] %></td>
                <td class="hidden"> <%= variant.pop_homs["Other"] %></td>
                <td class="hidden"> <%= variant.pop_acs["South Asian"] %></td>
                <td class="hidden"> <%= variant.pop_ans["South Asian"] %></td>
                <td class="hidden"> <%= variant.pop_homs["South Asian"] %></td>
            <% } %>
        </tr>
    <% }); %>
    </tbody>
</table>
</script>
